infra/terraform/modules/kubernetes_cluster/master.tf

```diff
@@ -0,0 +1,182 @@
+  vpc_zone_identifier  = ["${element(var.vpc_private_subnet_ids, count.index)}"]
+  launch_configuration = "${element(aws_launch_configuration.master.*.id, count.index)}"
+  load_balancers       = [
+    "${aws_elb.master.name}"
+  ]
+  max_size         = 1
+  min_size         = 1
+  desired_capacity = 1
+
+  tag = {
+    key                 = "KubernetesCluster"
+    value               = "${var.cluster_fqdn}"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "Name"
+    value               = "master-${data.aws_region.current.name}${element(split(",", data.template_file.az_letters.rendered), count.index)}.masters.${var.cluster_fqdn}"
+    propagate_at_launch = true
+  }
+
+  tag = {
+    key                 = "k8s.io/role/master"
+    value               = "1"
+    propagate_at_launch = true
+  }
+}
+
+resource "aws_elb" "master" {
+  name            = "master-${var.cluster_name}"
+  subnets         = ["${var.vpc_public_subnet_ids}"]
+  security_groups = [
+    "${aws_security_group.master_elb.id}",
+    "${var.sg_allow_http_s}"
+  ]
+  listener {
+    instance_port     = 443
+    instance_protocol = "tcp"
+    lb_port           = 443
+    lb_protocol       = "tcp"
+  }
+  health_check {
+    healthy_threshold   = 2
+    unhealthy_threshold = 2
+    timeout             = 3
+    target              = "TCP:443"
+    interval            = 30
+  }
+  tags {
+    Name              = "${var.cluster_name}_master"
+    KubernetesCluster = "${var.cluster_fqdn}"
+  }
+}
+
+resource "aws_route53_record" "master_elb" {
+  name = "api.${var.cluster_fqdn}"
+  type = "A"
+
+  alias = {
+    name                   = "${aws_elb.master.dns_name}"
+    zone_id                = "${aws_elb.master.zone_id}"
+    evaluate_target_health = false
+  }
+
+  zone_id = "/hostedzone/${var.route53_zone_id}"
+}
+
+resource "aws_security_group" "master" {
+  name        = "masters.${var.cluster_fqdn}"
+  vpc_id      = "${var.vpc_id}"
+  description = "${var.cluster_name} master"
+  tags = {
+    Name              = "masters.${var.cluster_fqdn}"
+    KubernetesCluster = "${var.cluster_fqdn}"
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_security_group_rule" "master_elb_to_master" {
+  type                     = "ingress"
+  security_group_id        = "${aws_security_group.master.id}"
+  source_security_group_id = "${aws_security_group.master_elb.id}"
+  from_port                = 443
+  to_port                  = 443
+  protocol                 = "tcp"
+}
+
+resource "aws_security_group" "master_elb" {
+  name        = "${var.cluster_name}-master-elb"
+  vpc_id      = "${var.vpc_id}"
+  description = "${var.cluster_name} master ELB"
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags {
+    Name = "${var.cluster_name}_master_elb"
+  }
+}
+
+data "template_file" "master_user_data" {
```

Comment:
This is really interesting, I didn't know TF had a `template_file` resource to render template files.